dd-trace-api/src/main/java/com/signalfx/tracing/api/TraceSetting.java

```diff
@@ -0,0 +1,69 @@
+package com.signalfx.tracing.api;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/** Use this annotation for class and method per-instrumentation configurations */
+@Retention(RUNTIME)
+@Target({TYPE, METHOD})
```

Comment:
I find it a bit strange to have this on `TYPE` (for classes). Having a hard time picturing a use-case for setting this at the class level and how that relates to transaction flow for tracing.

Reply:
For the allowed exceptions case, if you have a resource representing an endpoint with several http method processors, I think it would be preferable to annotate the class once instead of each method.  If some base resource is subclassed several times with the same desired whitelist, having to annotate each subclass would want to be avoided.

In general, I can see `TraceSetting` eventually being used to prevent specific classes from being instrumented without having to disable the instrumentation globally, so being able to annotate classes would be necessary for this case.